daemon/daemon_test.go

```diff
@@ -1197,8 +1178,23 @@ func (s *daemonSuite) TestRestartShutdown(c *check.C) {
 	rebootWaitTimeout = 100 * time.Millisecond
 	rebootNoticeWait = 150 * time.Millisecond
 
-	cmd := testutil.MockCommand(c, "shutdown", "")
-	defer cmd.Restore()
+	nRebootCall := 0
+	rebootCheck := func(ra boot.RebootAction, d time.Duration, ri *boot.RebootInfo) error {
+		// Check arguments passed to reboot call
+		nRebootCall++
+		c.Check(ra, check.Equals, boot.RebootReboot)
+		switch nRebootCall {
+		case 1:
+			c.Check(d, check.Equals, 100*time.Millisecond)
+			c.Check(ri, check.IsNil)
+		default:
```

Comment:
can you make this have cases for 1 and 2, and then make the default case fail the test in case we somehow have a bug calling reboot too many times?

Reply:
granted the check below will also fail the test but I like to see the mocked functions also validate this as it can make it more clear to see when a function is called an unexpected time what the arguments were when it was called unexpectedly

Reply:
Good point, I've added cases now.